crates/pipeline-manager/src/compiler/sql_compiler.rs

```diff
@@ -0,0 +1,533 @@
+
+    // Call executable with arguments
+    //
+    // In the future, it might be that flags can be passed to the SQL compiler through
+    // the program_config field of the pipeline.
+    let mut command = Command::new(sql_compiler_executable_file_path.clone());
+    command
+        .arg(input_sql_file_path.as_os_str())
+        .arg("-js")
+        .arg(output_json_schema_file_path.as_os_str())
+        .arg("-o")
+        .arg(output_rust_main_file_path.as_os_str())
+        .arg("-i")
+        .arg("-je")
+        .arg("--alltables")
+        .arg("--ignoreOrder")
+        .arg("--unquotedCasing")
+        .arg("lower")
+        .stdin(Stdio::null())
+        .stdout(Stdio::from(output_stdout_file.into_std().await))
+        .stderr(Stdio::from(output_stderr_file.into_std().await));
+
+    // Start process
+    let mut process = command.spawn().map_err(|e| {
+        SqlCompilationError::SystemError(
+            CommonError::io_error(
+                format!(
+                    "running SQL compiler executable '{}'",
+                    sql_compiler_executable_file_path.display()
+                ),
+                e,
+            )
+            .to_string(),
+        )
+    })?;
+
+    // Wait for process to exit while regularly checking if the pipeline still exists
+    // and has not had its program get updated
+    let exit_status = loop {
+        match process.try_wait() {
+            Ok(exit_status) => match exit_status {
+                None => {
+                    if let Some(db) = db.clone() {
+                        match db
+                            .lock()
+                            .await
+                            .get_pipeline_by_id(tenant_id, pipeline_id)
+                            .await
+                        {
+                            Ok(pipeline) => {
+                                if pipeline.program_version != program_version {
+                                    return Err(SqlCompilationError::Outdated);
+                                }
+                            }
+                            Err(DBError::UnknownPipeline { .. }) => {
+                                return Err(SqlCompilationError::Outdated);
+                            }
+                            Err(e) => {
+                                error!("SQL compilation outdated check failed due to database error: {e}")
+                                // As preemption check failing is not fatal, compilation will continue
+                            }
+                        }
+                    }
+                }
+                Some(exit_status) => break exit_status,
+            },
+            Err(e) => {
+                return Err(SqlCompilationError::SystemError(
+                    CommonError::io_error("waiting for SQL compilation process".to_string(), e)
+                        .to_string(),
+                ));
+            }
+        }
+        sleep(COMPILATION_CHECK_INTERVAL).await;
```

Comment:
same as above

Reply:
For this particular line, it checks the status of the process at an interval, as such this sleep seems appropriate?